### PR TITLE
fix(attendee registration shortcode): Correct asset enqueue for shortcode

### PR DESCRIPTION
**Pull request by Camwyn**
Ensure that the shortcode enqueues the correct assets when the toggle is off. Only on the short-coded
page. Only when the page is active (option set)

:ticket: https://central.tri.be/issues/125964 && https://central.tri.be/issues/126779

~On hold because I want to check into one more item before merging.~

### CODE REVIEW

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 1]
```diff
@@ -14,8 +14,42 @@ public function hook() {
...
+			return;
+		}
+
+		$slug = parse_url( $_SERVER[ 'REQUEST_URI' ], PHP_URL_PATH );
+		// not on a page we can parse
+		if ( ! $slug ) {
+			return;
+		}
+
+		$page = get_page_by_path( $slug, OBJECT );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]: Shouldn't this be available somewhere already on the current page, if it's a page? `get_queried_object_id()`?

File: src/resources/js/tickets-registration-page.js [hunk 2]
```diff
@@ -1,351 +1,355 @@
...
+	 * @since 4.10.1
+	 *
+	 * @return void
+	 */
+	obj.initContainers = function() {
+		$( obj.selector.container ).each( function() {
+			var $event = $( this );
+			var allRequired = obj.validateEventAttendees( $event );
+
+			if ( allRequired ) {
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: Was this intentional?

File: src/resources/js/tickets-registration-page.js [hunk 3]
```diff
@@ -1,351 +1,355 @@
...
+	/**
+	 * Handle save attendees info form submission via ajax.
+	 * Display a message if there are required fields missing.
+	 *
+	 * @since 4.9
+	 *
+	 * @return void
+	 */
+	obj.handleSaveSubmission = function( e ) {
+		console.log( 'handleSaveSubmission' );
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: Intentional?

File: src/resources/js/tickets-registration-page.js [hunk 4]
```diff
@@ -1,351 +1,355 @@
...
+	 * Update container status to complete
+	 *
+	 * @since 4.10.1
+	 *
+	 * @return void
+	 */
+	obj.updateStatusToComplete = function( $event ) {
+		var $statusObj = $event.find( obj.selector.status );
+		$statusObj.removeClass( 'incomplete' );
+		$statusObj.find( 'i' ).removeClass( 'dashicons-edit' );
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: If we are storing a var that needs to be reused, should we then store `$statusObj.find( 'i' )`?

File: src/resources/js/tickets-registration-page.js [hunk 5]
```diff
@@ -1,351 +1,355 @@
...
+	 * Update container status to incomplete
+	 *
+	 * @since 4.10.1
+	 *
+	 * @return void
+	 */
+	obj.updateStatusToIncomplete = function( $event) {
+		var $statusObj = $event.find( obj.selector.status );
+		$statusObj.addClass( 'incomplete' );
+		$statusObj.find( 'i' ).addClass( 'dashicons-edit' );
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: If we are storing a var that needs to be reused, should we then store `$statusObj.find( 'i' )`?

File: src/resources/js/tickets-registration-page.js (hunk 3 above)
Comment (Camwyn) [reply src/resources/js/tickets-registration-page.js]: nope

File: src/Tribe/Attendee_Registration/Shortcode.php (hunk 1 above)
Comment (Camwyn) [reply src/Tribe/Attendee_Registration/Shortcode.php]: you are correct sir... 🤦‍♂

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 6]
```diff
@@ -14,8 +14,36 @@ public function hook() {
...
+	public function maybe_enqueue_scripts() {
+		if (
+			is_archive()
+			|| is_admin()
+		) {
+			return;
+		}
+
+		$shortcode_page = tribe_get_option( 'ticket-attendee-page-id', false );
+		// option not set
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]:
```suggestion

		// Option is not set, don't enqueue scripts.
```

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 7]
```diff
@@ -14,8 +14,36 @@ public function hook() {
...
+		if ( ! $shortcode_page ) {
+			return;
+		}
+
+		$page = get_queried_object();
+		//not on the correct page
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]:
```suggestion

		// Not on a shortcode page, don't enqueue scripts.
```

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 8]
```diff
@@ -14,8 +14,36 @@ public function hook() {
...
 
+	public function maybe_enqueue_scripts() {
+		if (
+			is_archive()
+			|| is_admin()
+		) {
+			return;
+		}
+
+		$shortcode_page = tribe_get_option( 'ticket-attendee-page-id', false );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]:
```suggestion
		$shortcode_page = (int) tribe_get_option( 'ticket-attendee-page-id', false );
```

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 9]
```diff
@@ -14,8 +14,36 @@ public function hook() {
...
+		// option not set
+		if ( ! $shortcode_page ) {
+			return;
+		}
+
+		$page = get_queried_object();
+		//not on the correct page
+		if ( ! $page || absint( $shortcode_page ) !== absint( $page->ID ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]:
```suggestion
		if ( ! $page || ! $page instanceof WP_Post || $shortcode_page !== $page->ID ) {
```

**Pull request merged**